### DISCUSSION TITLE
Support for HTML custom elements tag

### DISCUSSION THREAD

**Discussion by mustzev** [tailwindlabs/tailwindcss#3512] [Ideas]
I find myself using a lot of div and span tags just so that I don't have to undo predefined styles of HTML semantic tags when using tailwind. (for example p tag comes with predefined margin style)

I think it would be great to add support for HTML custom elements tag just for the naming convention alone, since custom tags can be more descriptive than the current HTML semantic tags. Maybe I am not getting the whole web component thing correctly. What are you guys thoughts on this?

Or is it just an overdoing, when we could make do with the HTML semantic tags?

**Comment by richadr** [comment]
There is nothing stopping you from using custom tags.  Don't forget that they're `display: inline` by default.  You probably need to add `display: block` to most components.

Using html semantic tags is always a good idea.  They are the standard and improve accessibility of your app if you use it right.

You can control what predefined margins are set, see: https://tailwindcss.com/docs/preflight#default-margins-are-removed
